functions/src/analytics.ts

```diff
@@ -58,20 +61,33 @@ const getUsersWithVisits = async (venueIdsArray: string[]) => {
 };
 
 const generateAnalytics: HttpsFunctionHandler<{
-  venueIds: string[] | string | undefined;
+  worldSlug: string | undefined;
 }> = async (data, context) => {
   assertValidAuth(context);
 
-  if (!context.auth) return;
+  if (!context.auth) {
+    throw new HttpsError("internal", `No authentication context`);
+  }
 
   await checkIsAdmin(context.auth.token.user_id);
 
-  const venueIdsArray = Array.isArray(data?.venueIds)
-    ? data.venueIds
-    : (data?.venueIds ?? "").split(",");
+  const matchingVenues = await admin
+    .firestore()
+    .collection("venues")
+    .where("worldId", "==", data.worldSlug)
```

Comment:
```suggestion
    .where("slug", "==", data.worldSlug)
```

Reply:
changing worldId to slug doesn't make sense. I want to filter venues/spaces by worldId